core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,32 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for state in exploration.states.values():
+            if state.interaction.id == 'LogicProof':
+                owner_ids = rights_manager.get_exploration_rights(
+                    item.id).owner_ids
+                for user_id in owner_ids:
```

Comment:
What if `owner_ids` is `[]`? We should have a separate `yield` for this.

Reply:
Done yeild ('EMPTY', item.id) in that case